wear/src/main/java/com/eveningoutpost/dexdrip/Simulation.java

```diff
@@ -382,14 +375,14 @@ private void promptSpeechInput() {
     }
     private void handleWordPair() {
         boolean preserve = false;
-        if ((thisnumber == -1) || (thisword == "")) return;
+        if ((thisnumber == -1) || (Objects.equals(thisword, ""))) return;
```

Comment:
This doesn't look correct. Should it just be `thisword.equals("")` ? 

Here the compiler is already automatically fixing it to that to avoid the incorrect syntax of object instance comparison.

Reply:
Objects.equals(a,b) is one step up from a.equals(b) since it is null-safe. other possible variants are ["".equals(thisword)] or StringUtils.isEmpty(thisword)

Reply:
See line 70: String thisword = ""; and is never assigned to null. 
So thisword will always be initialized and never null. This change is strictly unneccessary?

Reply:
using "==" to compare string is not the correct way of comparing strings. I was guided to this line by codacy which suggested fixing it. If you really don't approve of it, I can revert the change, but given previous comments that the code has been build by a lot of different people over time, and the quality is variable, I find that fixing some of the issues standard tools like codacy suggests is a good thing.

Reply:
@jamorham Rewrote to "".equals(thisword)

Reply:
@jamorham, @dabear: regarding == vs .equals, there is a difference in java. Some of it is explained at stackoverflow: https://stackoverflow.com/questions/513832/how-do-i-compare-strings-in-java 
The most important difference is that == is comparing instances and .equals is comparing value. The compiler is not doing any optimization to the code, but the current way string is implemented is by using an internal cache in String to try to keep one instance of each string. The correct way to compare strings is to use either String.equels() or Objects.equals().